core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -122,17 +133,18 @@ public List<Result> scan(Scan scan) throws CrudException {
   private List<Result> scanInternal(Scan scan) throws CrudException {
     List<String> originalProjections = new ArrayList<>(scan.getProjections());
 
-    List<Result> results = new ArrayList<>();
+    Map<Snapshot.Key, TransactionResult> results = new LinkedHashMap<>();
 
-    Optional<List<Snapshot.Key>> keysInSnapshot = snapshot.get(scan);
-    if (keysInSnapshot.isPresent()) {
-      for (Snapshot.Key key : keysInSnapshot.get()) {
-        snapshot.get(key).ifPresent(results::add);
+    Optional<Map<Snapshot.Key, TransactionResult>> resultsInSnapshot = snapshot.get(scan);
+    if (resultsInSnapshot.isPresent()) {
+      for (Entry<Key, TransactionResult> entry : resultsInSnapshot.get().entrySet()) {
+        snapshot
+            .mergeResult(entry.getKey(), Optional.of(entry.getValue()))
```

Comment:
We do not use `mergeResult` with conjunctions for scans. We do not have to conjunction-aware merge with the write set since it will be considered in the write-after-scan check.